src/resources/trnmanager.c

```diff
@@ -3,26 +3,29 @@
 #include "formats/tournament.h"
 #include "resources/pathmanager.h"
 #include "utils/allocator.h"
-#include "utils/list.h"
 #include "utils/log.h"
 #include "utils/scandir.h"
+#include "utils/vector.h"
 #include <stdio.h>
 #include <string.h>
 
-static void trnlist_node_free_callback(void *data) {
-    sd_tournament_file *trn = data;
-    sd_tournament_free(trn);
+static int trn_sort_compare_fn(const void *a, const void *b) {
+    sd_tournament_file const *trn_a = a;
+    sd_tournament_file const *trn_b = b;
+    return (trn_a->registration_fee > trn_b->registration_fee) - (trn_a->registration_fee < trn_b->registration_fee);
```

Comment:
I *believe* the original game sorts by the ID first and then by the fee, if there's more than 1 with the same ID? See https://omf2097.com/wiki/doku.php?id=omf2097:robyrt_tourn

Reply:
thanks, i'll do a quick n' dirty TRN file edit to verify and then use the correct logic.

Reply:
the linked wiki is wrong. tournaments are not sorted by `TOURNAMENT NUMBER`.

Reply:
Good to know.